packages/media-fields/src/media_thumbnail/view.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import {
+	__experimentalTruncate as Truncate,
+	__experimentalVStack as VStack,
+	Icon,
+} from '@wordpress/components';
+import type { Attachment } from '@wordpress/core-data';
+import { getFilename } from '@wordpress/url';
+import type { DataViewRenderFieldProps } from '@wordpress/dataviews';
+/**
+ * Internal dependencies
+ */
+import { getMediaTypeFromMimeType } from '../utils/get-media-type-from-mime-type';
+import type { MediaItem } from '../types';
+
+export default function MediaThumbnailView( {
+	item,
+	config,
+}: DataViewRenderFieldProps< MediaItem > ) {
+	const _featuredMedia = useSelect(
+		( select ) => {
+			// Avoid the network request if it's not needed. `featured_media` is
+			// 0 for images and media without featured media.
+			if ( ! item.featured_media ) {
+				return;
+			}
+			return select( coreStore ).getEntityRecord< Attachment >(
+				'postType',
+				'attachment',
+				item.featured_media
+			);
+		},
+		[ item.featured_media ]
+	);
+	const featuredMedia = item.featured_media ? _featuredMedia : item;
+
+	// Fetching.
+	if ( ! featuredMedia ) {
+		return null;
+	}
+
+	const filename = getFilename( featuredMedia.source_url || '' );
+
+	if (
+		// Ensure the featured media is an image.
+		getMediaTypeFromMimeType( featuredMedia.mime_type ).type === 'image'
+	) {
+		return (
+			<div className="dataviews-media-field__media-thumbnail">
+				<img
```

Comment:
Idea for follow up. Loading state/error fallback? Hard to tell from the story. Maybe for later when we start using these in larger data views